config/application.rb

```diff
@@ -114,13 +114,6 @@ class Application < Rails::Application
     # on the top to bottom order of initializer calls in the file.
     # Because this is easy to mess up, keep your initializers in order.
     # register plugins even before loading settings, as plugins can bring their own settings
-    initializer :register_vmdb_plugins, :before => :load_vmdb_settings do
-      Rails.application.railties.each do |railtie|
-        next unless railtie.class.name.start_with?("ManageIQ::Providers::") || railtie.try(:vmdb_plugin?)
-        Vmdb::Plugins.instance.register_vmdb_plugin(railtie)
```

Comment:
calling `Vmdb::Plugins.instance` will register all plugins. So whoever calls the `.instance` first, will initialize it.
I think thats the nice improvement, that we dont need an initializer...

```
--> #0  Vmdb::Plugins.initialize at /Users/hild/src/manageiq/lib/vmdb/plugins.rb:13
    ͱ-- #1  Class.new(*args) at /Users/hild/.rbenv/versions/2.3.1/lib/ruby/2.3.0/singleton.rb:142
    #2  block in #<Class:Vmdb::Plugins>.block in instance at /Users/hild/.rbenv/versions/2.3.1/lib/ruby/2.3.0/singleton.rb:142
    ͱ-- #3  Thread::Mutex.synchronize at /Users/hild/.rbenv/versions/2.3.1/lib/ruby/2.3.0/singleton.rb:140
    #4  #<Class:Vmdb::Plugins>.instance at /Users/hild/.rbenv/versions/2.3.1/lib/ruby/2.3.0/singleton.rb:140
    #5  #<Class:Vmdb::Settings>.template_roots at /Users/hild/src/manageiq/lib/vmdb/settings.rb:118
    #6  #<Class:Vmdb::Settings>.template_sources at /Users/hild/src/manageiq/lib/vmdb/settings.rb:125
    #7  block in #<Class:Vmdb::Settings>.block in build_template at /Users/hild/src/manageiq/lib/vmdb/settings.rb:82
    ͱ-- #8  Kernel.tap at /Users/hild/src/manageiq/lib/vmdb/settings.rb:81
    #9  #<Class:Vmdb::Settings>.build_template at /Users/hild/src/manageiq/lib/vmdb/settings.rb:81
    #10 #<Class:Vmdb::Settings>.build_without_local(resource#Symbol) at /Users/hild/src/manageiq/lib/vmdb/settings.rb:109
    #11 #<Class:Vmdb::Settings>.build(resource#Symbol) at /Users/hild/src/manageiq/lib/vmdb/settings.rb:88
    #12 #<Class:Vmdb::Settings>.for_resource(resource#Symbol) at /Users/hild/src/manageiq/lib/vmdb/settings.rb:57
    #13 #<Class:Vmdb::Settings>.init at /Users/hild/src/manageiq/lib/vmdb/settings.rb:16
    #14 block in block in <class:Application> at /Users/hild/src/manageiq/config/application.rb:118
 ```